jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/MassIndexingJob.java

```diff
@@ -64,32 +65,34 @@ public ParametersBuilder forEntities(Class<?> rootEntity, Class<?>... rootEntiti
 		}
 	}
 
+	/**
+	 * Parameter builder for mass-indexing job. The default value of each parameter is defined in the job XML file
+	 * {@code hibernate-search-mass-indexing.xml}.
+	 */
 	public static class ParametersBuilder {
 
 		private final Set<Class<?>> rootEntities;
 		private String entityManagerFactoryScope;
 		private String entityManagerFactoryReference;
-		private boolean cacheable = false;
-		private boolean optimizeAfterPurge = false;
-		private boolean optimizeOnFinish = false;
-		private boolean purgeAllOnStart = false;
-		private int fetchSize = 200 * 1000;
-		private int checkpointInterval = 200;
-		private int rowsPerPartition = 250;
-		private int maxThreads = 1;
+		private Boolean cacheable;
+		private Boolean optimizeAfterPurge;
+		private Boolean optimizeOnFinish;
+		private Boolean purgeAllOnStart;
+		private Integer fetchSize;
+		private Integer checkpointInterval;
+		private Integer rowsPerPartition;
+		private Integer maxThreads;
 		private Set<Criterion> customQueryCriteria;
 		private String customQueryHql;
-		private int customQueryLimit = 1000 * 1000;
+		private Integer customQueryLimit;
 
 		private ParametersBuilder(Class<?> rootEntity, Class<?>... rootEntities) {
 			if ( rootEntity == null ) {
 				throw new IllegalArgumentException( "rootEntities must have at least 1 element." );
 			}
 			this.rootEntities = new HashSet<>();
 			this.rootEntities.add( rootEntity );
-			for ( Class<?> clz : rootEntities ) {
-				this.rootEntities.add( clz );
-			}
+			Collections.addAll( this.rootEntities, rootEntities );
 			customQueryCriteria = new HashSet<>();
 			customQueryHql = "";
```

Comment:
This should be removed, I think. Currently, this being non-null means that [we systematically add it to the parameters](https://github.com/yrodiere/hibernate-search/pull/4/files#diff-220e16d988e094ce942a55c1bf683410R293).
Be careful though, there are multiple places in the code where we expect `customQueryHql` to be non-null (we don't check for null-ness before calling `customQueryHql.isEmpty()` for instance)